src/viewport.js

```diff
@@ -1522,15 +1522,53 @@ $.Viewport.prototype = {
     },
 
     /**
-     * Toggle flip state and demands a new drawing on navigator and viewer objects.
-    */
+     * Toggles flip state and demands a new drawing on navigator and viewer objects.
+     * @function
+     * @return {OpenSeadragon.Viewport} Chainable.
+     */
     toggleFlip: function() {
-      this.flipped = !this.flipped;
+      this.setFlip(!this.getFlip());
```

Comment:
This line is great, but the rest of this function belongs in `setFlip` so that it happens regardless of whether the user calls this function or that.

Reply:
That's better, yes. Done! =)